TLS 1.2 Support When Installing PackageManagement Module

serkanz opened the pull request:
To download the PackageManagement module requires TLS 1.2. Many users will get an error. 

fixes #1314 

fixes https://github.com/PowerShell/vscode-powershell/issues/2763

TylerLeonhardt commented on src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs:
```diff
@@ -118,7 +118,7 @@ private async Task CheckPackageManagement()
                     StringBuilder errors = new StringBuilder();
                     await _powerShellContextService.ExecuteScriptStringAsync(
-                        "powershell.exe -NoLogo -NoProfile -Command 'Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber'",
+                        "powershell.exe -NoLogo -NoProfile -Command '[Net.ServicePointManager]::SecurityProtocol=[Net.SecurityProtocolType]::Tls12; Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber'",
```
```suggestion
                        "powershell.exe -NoLogo -NoProfile -Command '[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber' -Repository PSGallery",
```

TylerLeonhardt replied:
we got some other feedback about also including the Repository so can you also include that? Thanks!

TylerLeonhardt commented:
On line 143 there's a string that says:

```
PackageManagement update failed. Please run the following command in a new Windows PowerShell session and then restart the PowerShell extension: `Install-Module PackageManagement -Force -AllowClobber -MinimumVersion 1.4.6`
```

Can you change that to say:

```
PackageManagement update failed. This might be due to PowerShell Gallery using TLS 1.2. More info can be found at https://aka.ms/psgallerytls
```

serkanz commented:
Ok, updated the message string
> On line 143 there's a string that says:
> 
> ```
> PackageManagement update failed. Please run the following command in a new Windows PowerShell session and then restart the PowerShell extension: `Install-Module PackageManagement -Force -AllowClobber -MinimumVersion 1.4.6`
> ```
> 
> Can you change that to say:
> 
> ```
> PackageManagement update failed. This might be due to PowerShell Gallery using TLS 1.2. More info can be found at https://aka.ms/psgallerytls
> ```


Added **-Repository** argument
> Thanks! Couple small things to add

serkanz commented:
PowerShell/vscode-powershell#2763

TylerLeonhardt commented on src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs:
```diff
@@ -118,7 +118,7 @@ private async Task CheckPackageManagement()
                     StringBuilder errors = new StringBuilder();
                     await _powerShellContextService.ExecuteScriptStringAsync(
-                        "powershell.exe -NoLogo -NoProfile -Command 'Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber'",
+                        "powershell.exe -NoLogo -NoProfile -Command '[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber' -Repository PSGallery",
```
This command isn't quite right. You have the `'` before the `-Repository`

serkanz replied:
> This command isn't quite right. You have the `'` before the `-Repository`

Sorry, my mistake : ( will correct asap

SydneyhSmith commented:
fixes https://github.com/PowerShell/vscode-powershell/issues/2763
